File: src/level1/lotto.java
package level1;

import java.util.ArrayList;

public class lotto {
    public static int[] solution(int[] lottos, int[] win_nums) {
        int[] answer = new int[2];
        int match_num = 0;
        int unknown = 0;
        ArrayList<Integer> rest_nums = new ArrayList<>();
        for(int num : win_nums){
            rest_nums.add(num);
        }
        for (int num : lottos){
            if(rest_nums.contains(num)){
                rest_nums.remove(rest_nums.indexOf(num));
                match_num++;
            }
            if(num == 0){
                unknown++;
            }
        }

        if (unknown < 1 && match_num < 1){
            return new int[] {6,6};
        }

        if(unknown == 6){
            return new int[] {1,6};
        }

        if(match_num == 6){
            return new int[] {1,1};
        }else {
            answer[1] = 6 - match_num + 1;
            if (unknown <= rest_nums.size()){
                answer[0] = 6 - match_num - unknown + 1;
            }else {
                answer[0] = 6 - match_num - rest_nums.size() + 1;
            }

        }



        return answer;
    }


    public static void main(String[] args) {
        int[] lottos = {97, 96, 99, 98, 95, 94};
        int[] win_nums ={31, 10, 45, 1, 6, 19};
        int[] result = solution(lottos, win_nums);
        System.out.println(result[0] + ", " + result[1]);



    }
}

File: src/level1/lotto.kt
package level1

class lotto_kotlin {
    fun solution(lottos: IntArray, win_nums: IntArray): IntArray {
        var answer: IntArray = IntArray(2) {0}

        var unknown = 0
        var match_num = 0
        var rest_num = 6;
        for(i in lottos){
            if(i == 0){
                unknown++
            }else{
                if(win_nums.contains(i)){
                    match_num++
                }else{
                    rest_num--
                }
            }
        }
        if(unknown == 6){
            return IntArray(2) {1; 6}
        }
        if(match_num == 6){
            return IntArray(2){1;1}
        }
        if(match_num == 0 && unknown ==0){
            return IntArray(2){6;6}
        }
        answer[1] = 6 - match_num + 1
        if (unknown <= rest_num) {
            answer[0] = 6 - match_num - unknown + 1
        } else {
            answer[0] = 6 - match_num - rest_num + 1
        }
        return answer
    }
}